pyhf/modifiers/combined_mods.py

```diff
@@ -241,6 +240,8 @@ def __init__(self,shapefactor_mods,pdfconfig,mega_mods):
                 for s in pdfconfig.samples
             ] for m in shapefactor_mods
         ]
+        channel_bin_index_map = [j for c in pdfconfig.channels for j in range(pdfconfig.channel_nbins[c])]
```

Comment:
this tells us how to map the given parameter indices to the corresponding bins. For example..

```
[
 [c1b1,c1b2,c2b1c2b2,c2b3]  << signal
 [c1b1,c1b2,c2b1c2b2,c2b3]  << bkg1
 [c1b1,c1b2,c2b1c2b2,c2b3]  << background
]
```

ROOT will make nuispars: `gamma_shapefac_bin_1,gamma_shapefac_bin_2,gamma_shapefac_bin_3`. The 2 bins of the one channel will be affected by: `gamma_shapefac_bin_1,gamma_shapefac_bin_2` and the three bins of the other by: `gamma_shapefac_bin_1,gamma_shapefac_bin_2,gamma_shapefac_bin_3`

We need to count the number of bins per channel in the right order using `pdfconfig.channels` and generate indices for this using `range(num_bins_in_channel)`.

Reply:
this is not a look-up table, so probably another name than `map` might be good.. maybe `global_bin_numbers`, `global_bin_index`

Reply:
just making this concrete to help me understand.. this will result in e.g.

```
[
 [0,1,0,1,2]  << signal
 [0,1,0,1,2] << bkg1
 [0,1,0,1,2]  << background
]

```